tests/core/test_library.py

```diff
@@ -167,6 +170,16 @@ def test_lookup_uris_returns_empty_list_for_dummy3_track(self):
         self.assertFalse(self.library1.lookup.called)
         self.assertFalse(self.library2.lookup.called)
 
+    def test_lookup_adds_source(self):
+        # Overwrite the source_name attribute
+        self.backend1.source_name.get.return_value = 'hamspam'
+
+        track = Track(name='abc', source='hamspam')
+        self.library1.lookup().get.return_value = [track]
```

Comment:
`lookup()` should be `lookup.return_value` to not make an additional call to the mock.